### PR TITLE
fix: point.withinBBox() supports crossmeridian bounding boxes (#3994)

### LINKED ISSUE #3994: withinBBox doesn't seem to support crossmeridian
Hello !

```CYPHER
        WITH
          point({longitude: 179, latitude: 55.66}) AS lowerLeft,
          point({longitude: -179, latitude: 55.70}) AS upperRight
        RETURN point.withinBBox(point({longitude: 180, latitude: 55.66}), lowerLeft, upperRight) AS result
```

should return true

### PR DESCRIPTION
**Pull request by robfrank**
## Summary
- Fix `point.withinBBox()` returning false for points inside a bounding box that wraps the antimeridian (e.g. `lowerLeft.longitude=179`, `upperRight.longitude=-179`).
- Detect geographic coords (longitude/latitude) and apply OR-based longitude check when `lowerLeft.longitude > upperRight.longitude`. Cartesian (x/y) behavior is unchanged.

Fixes #3994

## Test plan
- [x] New unit tests in `CypherBuiltInFunctionsTest`: `pointWithinBBoxCrossmeridianInside`, `pointWithinBBoxCrossmeridianOutside`
- [x] New integration tests in `OpenCypherSpatialFunctionsComprehensiveTest` covering the exact Cypher query from the issue
- [x] All 13 `pointWithinBBox*` tests pass (9 pre-existing + 4 new), no regressions

### CODE REVIEW

File: engine/src/main/java/com/arcadedb/function/geo/PointWithinBBoxFunction.java [hunk 1]
```diff
@@ -44,13 +43,17 @@ public Object execute(final Object[] args, final CommandContext context) {
...
     final double[] lowerLeft = extractCoordinates(args[1]);
     final double[] upperRight = extractCoordinates(args[2]);
-    return lowerLeft[0] <= point[0] && point[0] <= upperRight[0]
-        && lowerLeft[1] <= point[1] && point[1] <= upperRight[1];
+    final boolean longitudeInRange;
+    if (isGeographic(args[1]) && lowerLeft[0] > upperRight[0])
+      // Crossmeridian bbox: lowerLeft.longitude > upperRight.longitude means the box wraps around ±180
+      longitudeInRange = point[0] >= lowerLeft[0] || point[0] <= upperRight[0];
+    else
+      longitudeInRange = lowerLeft[0] <= point[0] && point[0] <= upperRight[0];
```
[the remark this answers is not in the document]
Comment (robfrank) [reply engine/src/main/java/com/arcadedb/function/geo/PointWithinBBoxFunction.java:56]: Implemented. Added the three-way `isGeographic` consistency check at the top of `execute()` — if `point`, `lowerLeft`, and `upperRight` are not all geographic or all Cartesian, returns `null` per the Cypher spec.

Note: I made `isGeographic` read the `crs` field (which `point()` always sets to `WGS-84*` or `cartesian*`), with key-based detection only as a fallback for raw user maps. This is needed because `CypherPointFunction` always emits both `x/y` and `longitude/latitude` keys for WGS-84 points, so `containsKey("longitude")` alone cannot distinguish CRS reliably.

Commit: 240f40696

File: engine/src/main/java/com/arcadedb/function/geo/PointWithinBBoxFunction.java [hunk 2]
```diff
@@ -59,4 +62,8 @@ private double[] extractCoordinates(final Object value) {
     }
     throw new CommandExecutionException("point.withinBBox() arguments must be point values (maps with x/y or longitude/latitude)");
   }
+
+  private boolean isGeographic(final Object value) {
+    return value instanceof Map<?, ?> map && map.containsKey("longitude");
```
[the remark this answers is not in the document]
Comment (robfrank) [reply engine/src/main/java/com/arcadedb/function/geo/PointWithinBBoxFunction.java]: Good catch — `extractCoordinates` did pick `x/y` first while `isGeographic` only looked for `longitude`, so a map carrying both keys could be read inconsistently.

Fixed by routing both through a shared `isGeographic`:
- `isGeographic` reads `crs` first (WGS-84 vs cartesian), with key inspection only as fallback for raw maps without `crs`.
- `extractCoordinates` consults `isGeographic` upfront and pulls `longitude/latitude` for geographic Points, even when `x/y` are also present.

This was important because `point()` always emits `x/y` mirrors of `longitude/latitude` on WGS-84 Points (see `CypherPointFunction#execute`), so a strict "x/y wins" rule would skip the crossmeridian branch on every `point({longitude:…, latitude:…})` value.

Commit: 240f40696

**Comment by robfrank** [comment]
Addressing review feedback in 240f40696.

**Implemented**
- **Gemini (line 51)** — mixed coordinate systems now return `null` per Cypher spec (three-way `isGeographic` check on point/lowerLeft/upperRight).
- **Gemini (line 67)** — `isGeographic` and `extractCoordinates` now use a single source of truth. `isGeographic` reads the `crs` field first (which `point()` always sets to `WGS-84*` or `cartesian*`), with key inspection only as a fallback for raw user maps. `extractCoordinates` consults `isGeographic` so a Point with both `x/y` and `longitude/latitude` (the normal output of `point({longitude:…, latitude:…})`) is read as geographic.
- **Claude #2** — added `pointWithinBBoxCrossmeridianLongitudeInsideLatitudeOutside`: longitude inside the crossmeridian arc, latitude out of range → `false`.
- **Claude #4** — added `pointWithinBBoxCrossmeridianAtAntimeridian` exercising both `+180.0` and `-180.0`.
- Added `pointWithinBBoxMixedCoordinateSystemsReturnsNull` covering the Cypher-spec null return.

**Not changed (with reasoning)**
- **Claude #1** (`isGeographic` only checking `args[1]`) — subsumed by Gemini's three-way check; once all three arguments are validated for system consistency, only one detection result matters.
- **Claude #3** (latitude boundary value clarity in the existing crossmeridian test) — `pointWithinBBoxOnEdge` already covers inclusive-boundary semantics for both axes; rewriting the crossmeridian test to use a strictly interior latitude is style preference, not coverage.

**Failing CI**
- `integration-tests`: `RaftServerDatabaseBackupIT.sqlBackup` — unrelated HA Raft test, this PR only touches the geo function and its tests.
- `Meterian client scan`: external scanner job failure, not actionable in this PR.
- `codecov/patch`: improved by the new tests.
